x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/ClassificationInferenceResults.java

```diff
@@ -0,0 +1,155 @@
+
+public class ClassificationInferenceResults extends SingleValueInferenceResults {
+
+    public static final String RESULT_TYPE = "classification";
+    public static final ParseField CLASSIFICATION_LABEL = new ParseField("classification_label");
+    public static final ParseField TOP_CLASSES = new ParseField("top_classes");
+    
+    private final String classificationLabel;
+    private final List<TopClassEntry> topClasses;
+
+    public ClassificationInferenceResults(double value, String classificationLabel, List<TopClassEntry> topClasses) {
+        super(value);
+        this.classificationLabel = classificationLabel;
+        this.topClasses = topClasses == null ? null : Collections.unmodifiableList(topClasses);
+    }
+
+    public ClassificationInferenceResults(StreamInput in) throws IOException {
+        super(in);
+        this.classificationLabel = in.readOptionalString();
+        if (in.readBoolean()) {
+            this.topClasses = Collections.unmodifiableList(in.readList(TopClassEntry::new));
+        } else {
+            this.topClasses = null;
+        }
+    }
+
+    public String getClassificationLabel() {
+        return classificationLabel;
+    }
+
+    public List<TopClassEntry> getTopClasses() {
+        return topClasses;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeOptionalString(classificationLabel);
+        out.writeBoolean(topClasses != null);
+        if (topClasses != null) {
+            out.writeCollection(topClasses);
+        }
+    }
+
+    @Override
+    XContentBuilder innerToXContent(XContentBuilder builder, Params params) throws IOException {
+        if (classificationLabel != null) {
+            builder.field(CLASSIFICATION_LABEL.getPreferredName(), classificationLabel);
+        }
+        if (topClasses != null) {
+            builder.field(TOP_CLASSES.getPreferredName(), topClasses);
+        }
+        return builder;
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == this) { return true; }
+        if (object == null || getClass() != object.getClass()) { return false; }
+        ClassificationInferenceResults that = (ClassificationInferenceResults) object;
+        return Objects.equals(value(), that.value()) &&
+            Objects.equals(classificationLabel, that.classificationLabel) &&
+            Objects.equals(topClasses, that.topClasses);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(value(), classificationLabel, topClasses);
+    }
+
+    @Override
+    public String resultType() {
+        return RESULT_TYPE;
+    }
+
+    @Override
+    public String valueAsString() {
+        return classificationLabel == null ? super.valueAsString() : classificationLabel;
+    }
+
+    public static class TopClassEntry implements ToXContentObject, Writeable {
+
+        public final ParseField LABEL = new ParseField("label");
```

Comment:
Should we consider calling this `class`?

Reply:
I opted for `classification` seems to fit better and prevents funky clashes with the `class` keyword in Java :)